src/test/java/org/opensearch/sdk/TestNamedWriteableRegistryAPI.java

```diff
@@ -135,12 +135,12 @@ public void testInvalidCategoryClass() throws UnknownHostException, IOException
         StreamInput in = new InputStreamStreamInput(input);
 
         // Category Class ExtensionRunner is not registered
-        NamedWriteableRegistryParseRequest request = new NamedWriteableRegistryParseRequest(ExtensionsRunner.class, in);
+        NamedWriteableRegistryParseRequest request = new NamedWriteableRegistryParseRequest(NamedWriteable.class, in);
```

Comment:
(Opinion, ok but think about changing) I think the intention of this test is to have an invalid class that will never be a NamedWriteable.  This probably works as that's an interface, but I'd prefer to use a class we have more control over on the SDK side. that we know will never be a registered namedWritable (per the comment above on line 137).

Reply:
I have a question on it. Since we added generic type for `NamedWriteableRegistryParseRequest`, if we use an invalid class which will never be a NamedWriteable instead of NameWriteable, the object `request` will not able to be generated. It will cause error like `incompatible types`.

Reply:
Ah, that makes sense because that is the whole reason for you adding the type safety.

But we know we will not ever register the interface in the registry, so this is OK as you have it!